kaldb/src/main/java/com/slack/kaldb/chunkManager/ChunkManager.java

```diff
@@ -40,16 +36,21 @@
   // TODO: We want to move this to the config eventually
   // Less than KaldbDistributedQueryService#READ_TIMEOUT_MS
   public static final int QUERY_TIMEOUT_SECONDS = 10;
-  public static final int LOCAL_QUERY_THREAD_POOL_SIZE = 4;
+  public static final int MIN_QUERY_THREAD_POOL_SIZE = 20;
+  public static final int MAX_QUERY_THREAD_POOL_SIZE = 200;
 
   private static final ExecutorService queryExecutorService = queryThreadPool();
 
   /*
      One day we will have to think about rate limiting/backpressure and we will revisit this so it could potentially reject threads if the pool is full
   */
   private static ExecutorService queryThreadPool() {
-    return Executors.newFixedThreadPool(
-        LOCAL_QUERY_THREAD_POOL_SIZE,
+    return new ThreadPoolExecutor(
+        MIN_QUERY_THREAD_POOL_SIZE,
+        MAX_QUERY_THREAD_POOL_SIZE,
+        60L,
+        TimeUnit.SECONDS,
+        new LinkedBlockingQueue<>(),
```

Comment:
This should probably use a `SynchronousQueue`, or alternatively just use the `Executors.newCachedThreadPool`. Using an unbounded queue to back this executor will make it harder to debug, can be susceptible to lockups. 

The recommended default queuing strategy is to use a `SynchronousQueue` unless there is a specific need for one of the alternative queueing strategies.